test/parallel/test-fs-readfile-flags.js

```diff
@@ -0,0 +1,70 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+'use strict';
+
+// Test of fs.readFile with different flags.
+const common = require('../common');
+const fs = require('fs');
+const assert = require('assert');
+const fixtures = require('../common/fixtures');
+
+const fn = fixtures.path('empty.txt');
+const nonexistingFn = fixtures.path('nonexistingfile.txt');
+const nonexistingFn2 = fixtures.path('nonexistingfile2.txt');
+
+fs.readFile(
+  fn,
+  // With `a+` the file is created if it does not exist
+  { encoding: 'utf8', flag: 'a+' },
+  common.mustCall((err, data) => {
+    assert.strictEqual(data, '');
+  })
+);
+
+fs.readFile(
+  fn,
+  // Like `a+` but fails if the path exists.
+  { encoding: 'utf8', flag: 'ax+' },
+  common.mustCall((err, data) => {
+    assert.strictEqual(err.code, 'EEXIST');
+  })
+);
+
+fs.readFile(
+  nonexistingFn,
+  // With `a+` the file is created if it does not exist
+  { encoding: 'utf8', flag: 'a+' },
+  common.mustCall((err, data) => {
+    assert.strictEqual(data, '');
+
+    fs.unlinkSync(nonexistingFn);
+  })
+);
+
+fs.readFile(
+  nonexistingFn2,
+  // Default flag is `r`. An exception occurs if the file does not exist.
+  { encoding: 'utf8' },
+  common.mustCall((err, data) => {
+    assert.strictEqual(err.code, 'ENOENT');
+  })
+);
```

Comment:
The fixtures module will work, but in this case (where we are creating files in the test and our only need is an empty file), it is better to use [tmpdir](https://github.com/nodejs/node/tree/9687348577347cf41f10617a6b3b457825347cf2/test/common#tmpdir-module).

```suggestion
'use strict';

// Test of fs.readFile with different flags.
const common = require('../common');
const fs = require('fs');
const assert = require('assert');
const path = require('path');
const tmpdir = require('../common/tmpdir');

tmpdir.refresh();

{
  const emptyFile = path.join(tmpdir.path, 'empty.txt');
  fs.closeSync(fs.openSync(emptyFile, 'w'));

  fs.readFile(
    emptyFile,
    // With `a+` the file is created if it does not exist
    { encoding: 'utf8', flag: 'a+' },
    common.mustCall((err, data) => { assert.strictEqual(data, ''); })
  );

  fs.readFile(
    emptyFile,
    // Like `a+` but fails if the path exists.
    { encoding: 'utf8', flag: 'ax+' },
    common.mustCall((err, data) => { assert.strictEqual(err.code, 'EEXIST'); })
  );
}

{
  const willBeCreated = path.join(tmpdir.path, 'will-be-created');

  fs.readFile(
    willBeCreated,
    // With `a+` the file is created if it does not exist
    { encoding: 'utf8', flag: 'a+' },
    common.mustCall((err, data) => { assert.strictEqual(data, ''); })
  );
}

{
  const willNotBeCreated = path.join(tmpdir.path, 'will-not-be-created');

  fs.readFile(
    willNotBeCreated,
    // Default flag is `r`. An exception occurs if the file does not exist.
    { encoding: 'utf8' },
    common.mustCall((err, data) => { assert.strictEqual(err.code, 'ENOENT'); })
  );
}
```

Reply:
Ok, I'll use your suggestion then, thank you!